### DISCUSSION TITLE
Force rescan of photos

### DISCUSSION THREAD

**Discussion by Supetorus** [deckerst/aves#897] [Q&A]
Is there a way to force a rescan of all my photos? I moved some photos between directories on another machine and synchronized my photo vault with syncthing. The photos still show in the location they were in before. I suspect this is because I moved the photos with digikam and it didn't update the modification timestamp so aves doesn't see they have changed. Is there a way to force aves to rescan my collection to fix this?

Edit: I have closed and opened aves multiple times and have also confirmed in cx file explorer that the images are where I think they are. I would imagine reinstalling would accomplish a rescan but that's pretty tedious to do every time organize files outside the app.

**Comment by deckerst** [comment]
Nope, this is up to the Android Media Store. Aves only gets access to public media on the device, and the media list comes from the native Media Store.

It will eventually discover the media, but the best is to use apps that properly create and move media files around, so that the Media Store knows about them right away.

**Reply by Supetorus** [reply]
Wow, well that is pretty unfortunate. I moved hundreds of file almost a full day ago and it still hasn't updated. Thanks for the response though.

**Reply by deckerst** [reply]
That seems pretty long. Maybe a device reboot would help.

**Reply by Supetorus** [reply]
Yep. Restarted and went straight to aves. Initially it showed no change then message showed cataloging and it all updated.

It's interesting that cx file explorer showed the files in the correct place but aves didn't. They must use a different method of loading information than the android media store, though I'm guessing it would be more work.

Just want to say as well that I really appreciate all the work you've put into this app. I read in another comment somewhere that you do this as a hobby or side project, yet somehow it's the best photo app for Android out of any I've seen, and not only that but there's no monetization that I've seen, not even a donate button in the app. Well done friend.

**Reply by deckerst** [reply]
File manager apps don't have the same limitation as gallery apps. The permission to access all files is considered overreach for galleries. That's a policy of the Play Store with which I agree.

Thanks for the kind words! Yes I do that in my spare time. As I don't do this for money or fame, I'm free to explore ideas and implement whatever I want. Fun for me, and hopefully useful to some people ^^

**Reply by Supetorus** [reply]
Ah, that makes sense. I don't know anything about android development so I just assumed that apps were only limited in permissions by what they request and are granted by the user. I suppose you could put it on other stores but I'm sure every store has its own associated headache.

It would be cool if you considered PRs, though I get wanting full ownership of every part. I've wanted to check out Flutter and contributing to a cool project like this would be a nice place to start. I'll have to see how you've licensed it, perhaps I could fork and build for my own phone.
